test/e2e/specs/plugins-api.test.js

```diff
@@ -27,6 +30,31 @@ describe( 'Using Plugins API', () => {
 		} );
 	} );
 
+	describe( 'Publish Panel', () => {
+		afterEach( async () => {
+			// Close Publish panel.
+			await page.click( '.editor-post-publish-panel__header button[aria-label="Close panel"]' );
+		} );
+
+		it( 'Should render publish panel inside Pre-publish sidebar', async () => {
+			// Type something first to activate Publish button.
+			await clickBlockAppender();
+			await page.keyboard.type( 'First paragraph' );
+
+			await openPublishPanel();
+
+			const pluginPublishPanelText = await page.$eval( '.editor-post-publish-panel .my-publish-panel-plugin__pre', ( el ) => el.innerText );
+			expect( pluginPublishPanelText ).toMatch( 'My pre publish panel' );
+		} );
+
+		it( 'Should render publish panel inside Post-publish sidebar', async () => {
```

Comment:
I think this test may have a dependency on the previous test. It probably needs the paragraph added in the previous test so the publish button can be clicked. Maybe we can just do beforeEach newPost and add the paragraph in both tests if we want to remove the dependency. Watching to the case I don't think in this particular situation the dependency is problematic.

Reply:
I don’t remember this use case, can tweak as suggested. It makes sense what you said 👍

Reply:
f2d5e89 - much better, I like it 💯